### PR TITLE
Fixes #178 - Unsupported ARIA attribute fails with global attributes

### LINKED ISSUE #178: UnsupportedAriaAttribute audit test fails when input has aria-label
The following HTML results in AX_ARIA_10 rule (unsupported aria attribute) failing.  This attribute should be valid.

```javascript
<input aria-label="my test" type="text">
```

This is the first time I've looked at these tools so I am not very familiar with the project, but the `attributeName` check in the following line of code looks suspect to me:

```javascript
if (axs.constants.ARIA_PROPERTIES.hasOwnProperty(lookupName) && !(attributeName in supported)) {
    return true;
}
```

### PR DESCRIPTION
**Pull request by RickSBrown**
```supported``` was sometimes an object, sometimes an array.
Probably best as an object for fast lookup.

### CODE REVIEW

**Comment by RickSBrown** [comment]
@alice PTAL

File: src/audits/UnsupportedAriaAttribute.js [hunk 1]
```diff
@@ -49,7 +51,15 @@ goog.require('axs.utils');
             }
             else {
                 // This test ignores the fact that some HTML elements should not take even global attributes.
-                supported = axs.constants.GLOBAL_PROPERTIES;
+                if(globalProperties) {  // if we have previously computed this value, use it
```
Comment (alice) [inline src/audits/UnsupportedAriaAttribute.js]: This is nice, but I think it's an unnecessary optimisation. Other initialisation happens right inside Constants.js, e.g. https://github.com/GoogleChrome/accessibility-developer-tools/blob/master/src/js/Constants.js#L982 - could you use this pattern instead?

File: test/audits/unsupported-aria-attribute-test.js [hunk 2]
```diff
@@ -125,4 +125,12 @@
         deepEqual(rule.run({ scope: fixture }), expected);
     });
 
+	test('Element with no role and global aria- attributes', function() {
```
Comment (alice) [inline test/audits/unsupported-aria-attribute-test.js]: Nice test :)

**Comment by RickSBrown** [comment]
@alice hopefully it's a bit more consistent with other constants now... Happy to tweak it further if required.

**Comment by RickSBrown** [comment]
@googlebot alice is the author of those commits and has signed the CLA / works for Google.

**Comment by RickSBrown** [comment]
alice (the project owner) is the author of the other commits and has signed
the CLA / works for Google

**Comment by alice** [comment]
:+1:

**Pull request merged**